Modern/wrapperPlugins/src/main/java/org/bonej/wrapperPlugins/wrapperUtils/UsageReporterOptions.java

```diff
@@ -0,0 +1,135 @@
+public class UsageReporterOptions extends ContextCommand {
+
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message1 = "Allow usage data collection?";
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message2 = "BoneJ would like to collect data on";
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message3 = "which plugins are being used, to direct development"; 
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message4 = "and promote BoneJ to funders.";
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message5 = "If you agree to participate please check the box.";		
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String helpMessage = "For more information click Help.";
+	
+	@Parameter(label = "Opt in to usage data collection", description = "Can BoneJ send usage data?")
+  private boolean optIn = false;
+	
+	@Parameter(label = "Help")
+	private Button button;
+	
+	@Parameter
+	private ImageJ imagej;
+
+	/** set to true if user permission has been requested */
+	static final String OPTINSET = "bonej2.report.option.set";
+	/** Set to false if reporting is not allowed */
+	static final String OPTINKEY = "bonej2.allow.reporter";
+	static final String COOKIE = "bonej2.report.cookie";
+  static final String COOKIE2 = "bonej2.report.cookie2";
+  /** time of first visit in seconds */
+	static final String FIRSTTIMEKEY = "bonej2.report.firstvisit";
+	/** unique ID for this particular BoneJ session */
+	static final String SESSIONKEY = "bonej2.report.bonejsession";
+	static final String IJSESSIONKEY = "bonej2.report.ijsession";
+
+	@Override
+	public void run() {
+
+		//TODO link to detailed online info
+    //For more information click Help.";
+		
+		//Wipe persistent data on opt-out
+		if (!optIn) {
+			System.out.print("User has opted out of data collection\n");
+			imagej.prefs().clear(this.getClass());
+		}
+
+		else {
+			System.out.print("User has opted in to data collection\n");
+			imagej.prefs().put(this.getClass(), OPTINKEY, true);
+			imagej.prefs().put(this.getClass(), COOKIE,
+				new Random().nextInt(Integer.MAX_VALUE));
+			imagej.prefs().put(this.getClass(), COOKIE2,
+				new Random().nextInt(Integer.MAX_VALUE));
+			imagej.prefs().put(this.getClass(), FIRSTTIMEKEY,
+				System.currentTimeMillis() / 1000);
+			imagej.prefs().put(this.getClass(), SESSIONKEY, 1);
+			imagej.prefs().put(this.getClass(), IJSESSIONKEY, 1);
+		}
+
+		//set that user permissions have been sought
+		System.out.println("User permission has been sought.\n");
+		imagej.prefs().put(this.getClass(), OPTINSET, true);
+		
+		System.out.println("URO Sending usage report...");
+		UsageReporter.reportEvent(this).send();
+	}
+	
+	/**
+	 * Check whether user has given permission to collect usage data
+	 * 
+	 * @return true only if the user has given explicit permission to send usage data
+	 */
+	public boolean isAllowed() {
```

Comment:
Like I said earlier, `isAllowed()` should be a private method of `UsageReporter`. This command just handles preferences, it doesn't act on them.

Reply:
OK - fixed in d31c7564